pkg/updatestrategy/node_pool_manager.go

```diff
@@ -120,6 +121,10 @@ func (m *KubernetesNodePoolManager) GetPool(ctx context.Context, nodePoolDesc *a
 
 	for _, npNode := range nodePool.Nodes {
 		if node, ok := instanceIDMap[npNode.ProviderID]; ok {
+			role := WorkerNodeRole
+			if r, ok := node.Annotations[nodeRoleAnnotaton]; ok && NodeRole(r) == MasterNodeRole {
```

Comment:
We have `NodePool.IsMaster()`, could we use that one instead?

Reply:
Changed to decide node type based on the node pool